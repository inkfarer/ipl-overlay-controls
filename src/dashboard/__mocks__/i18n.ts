import i18next from 'i18next';
import { config } from '@vue/test-utils';

jest.mock('i18next-vue', () => ({
    __esModule: true,
    useTranslation: () => ({ t: i18next.t })
}));

beforeAll(async () => {
    await i18next.init({
        lng: 'cimode',
        appendNamespaceToCIMode: true
    });
    config.global.mocks = {
        $t: i18next.t
    };
});
