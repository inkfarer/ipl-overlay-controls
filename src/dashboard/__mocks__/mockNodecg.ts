import Mock = jest.Mock;

export type ReplicantChangeHandler = (newValue?: unknown, oldValue?: unknown) => void;
export type MockReplicant = { value: unknown };
type OnFunction = (event: string, handler: ReplicantChangeHandler) => void;

export class MockNodecg {
    replicants: {[key: string]: MockReplicant};
    listeners: {[key: string]: ReplicantChangeHandler};
    sendMessage: Mock

    constructor() {
        this.replicants = {};
        this.listeners = {};
        this.sendMessage = jest.fn();
    }

    init(): void {
        const self = this;
        window.nodecg = {
            // @ts-ignore: Just a mock
            Replicant(name: string) {
                const replicantValue: { value: unknown, on: OnFunction } = {
                    value: undefined,
                    on: (event: string, handler: ReplicantChangeHandler) => {
                        self.listeners[name] = handler;
                    }
                };
                self.replicants[name] = replicantValue;
                return replicantValue;
            },
            sendMessage: self.sendMessage,
            // @ts-ignore: Just a mock
            log: {
                info: jest.fn()
            }
        };

        // @ts-ignore: It's a mock.
        window.NodeCG = {
            waitForReplicants: jest.fn().mockResolvedValue(true)
        };
    }
}
