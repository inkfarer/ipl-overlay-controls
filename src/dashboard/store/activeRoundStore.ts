import { ActiveRound, SwapColorsInternally } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { GameWinner } from 'types/enums/gameWinner';
import { SetActiveColorRequest, SetWinnerRequest, SwapRoundColorRequest } from 'types/messages/activeRound';
import { SetRoundRequest } from 'types/messages/rounds';
import { ActiveRoundGame } from 'types/activeRoundGame';

const activeRound = nodecg.Replicant<ActiveRound>('activeRound');
const swapColorsInternally = nodecg.Replicant<SwapColorsInternally>('swapColorsInternally');

export const activeRoundReps = [ activeRound, swapColorsInternally ];

export interface ActiveRoundStore {
    activeRound: ActiveRound;
    swapColorsInternally: SwapColorsInternally;
}

export const activeRoundStore = createStore<ActiveRoundStore>({
    state: {
        activeRound: null,
        swapColorsInternally: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        setWinner(store, { winner }: { winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner });
        },
        removeWinner(): void {
            nodecg.sendMessage('removeWinner');
        },
        setActiveColor(store, request: SetActiveColorRequest): void {
            nodecg.sendMessage('setActiveColor', request);
        },
        swapColors(): void {
            swapColorsInternally.value = !swapColorsInternally.value;
        },
        setWinnerForIndex(store, { index, winner }: { index: number, winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner, roundIndex: index } as SetWinnerRequest);
        },
        resetActiveRound(): void {
            nodecg.sendMessage('resetActiveRound');
        },
        setActiveRound(store, request: SetRoundRequest): void {
            nodecg.sendMessage('setActiveRound', request);
        },
        swapRoundColor(store, request: SwapRoundColorRequest): void {
            nodecg.sendMessage('swapRoundColor', request);
        },
        updateActiveGames(store, games: ActiveRoundGame[]): void {
            nodecg.sendMessage('updateActiveGames', { games });
        },
        switchToNextColor(): void {
            nodecg.sendMessage('switchToNextColor');
        },
        switchToPreviousColor(): void {
            nodecg.sendMessage('switchToPreviousColor');
        },
        async getNextAndPreviousColors(): Promise<void> {
            return nodecg.sendMessage('getNextAndPreviousColors');
        }
    }
});

export const activeRoundStoreKey: InjectionKey<Store<ActiveRoundStore>> = Symbol();

export function useActiveRoundStore(): Store<ActiveRoundStore> {
    return useStore(activeRoundStoreKey);
}
