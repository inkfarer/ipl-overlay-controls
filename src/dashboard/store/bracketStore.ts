import { Bracket } from '@tourneyview/common';
import { defineStore } from 'pinia';

const bracketData = nodecg.Replicant<Bracket | undefined>('bracketData');

export const bracketReps = [ bracketData ];

export interface BracketStore {
    bracketData: Bracket | undefined;
}

export const useBracketStore = defineStore('brackets', {
    state: () => ({
        bracketData: undefined
    } as BracketStore),
    actions: {
        resetBracketData() {
            bracketData.value = undefined;
        }
    }
});
