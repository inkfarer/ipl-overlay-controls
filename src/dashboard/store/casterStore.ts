import { NodeCGBrowser } from 'nodecg/browser';
import { Caster, Casters, RadiaSettings } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { generateId } from '../../helpers/generateId';

const casters = nodecg.Replicant<Casters>('casters');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');

export const castersReps = [ casters, radiaSettings ];

export interface CasterStore {
    casters: Casters;
    uncommittedCasters: Casters;
    radiaSettings: RadiaSettings;
}

const defaultCaster = {
    name: 'New Caster',
    twitter: '',
    pronouns: '?/?'
};

export const casterStore = createStore<CasterStore>({
    state: {
        casters: {},
        uncommittedCasters: {},
        radiaSettings: {
            enabled: null,
            guildID: null,
            updateOnImport: null
        }
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        },
        updateCaster(store, { id, newValue }: { id: string, newValue: Caster }): void {
            casters.value[id] = newValue;
        },
        addUncommittedCaster(store, { id, caster }: { id: string, caster?: Caster }): void {
            store.uncommittedCasters[id] = caster ?? defaultCaster;
        },
        removeUncommittedCaster(store, id: string): void {
            delete store.uncommittedCasters[id];
        }
    },
    actions: {
        addUncommittedCaster(store): string {
            const id = generateId();
            store.commit('addUncommittedCaster', { id });
            return id;
        },
        async saveUncommittedCaster(store, { id, caster }: { id: string, caster: Caster }): Promise<string> {
            const newId = await nodecg.sendMessage('saveCaster', caster);
            store.commit('removeUncommittedCaster', id);
            return newId;
        },
        async loadCastersFromVc(store): Promise<void> {
            const result = await nodecg.sendMessage('getLiveCommentators');
            if (result.extra && result.extra.length > 0) {
                for (let i = 0; i < result.extra.length; i++) {
                    const extraCaster = result.extra[i];
                    const id = extraCaster.discord_user_id;
                    delete extraCaster.discord_user_id;
                    store.commit('addUncommittedCaster', { id, caster: extraCaster });
                }
            }
        },
        showCasters() {
            nodecg.sendMessage('mainShowCasters');
        },
        removeCaster(store, id: string): void {
            nodecg.sendMessage('removeCaster', { id });
        }
    }
});

export const casterStoreKey: InjectionKey<Store<CasterStore>> = Symbol();

export function useCasterStore(): Store<CasterStore> {
    return useStore(casterStoreKey);
}
