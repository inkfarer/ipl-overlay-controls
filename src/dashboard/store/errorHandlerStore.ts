import { createStore, Store, useStore } from 'vuex';
import { App, InjectionKey } from 'vue';
import { generateId } from '../../helpers/generateId';

export interface ErrorHandlerStore {
    recentErrors: Record<string, unknown>;
}

export const errorHandlerStore = createStore<ErrorHandlerStore>({
    state: {
        recentErrors: {}
    },
    mutations: {
        removeRecentError(state, { key }: { key: string }): void {
            delete state.recentErrors[key];
        }
    },
    actions: {
        handleError(store, { err, info }: { err: unknown, info: string }): void {
            console.error(`Got error from '${info}': \n`, err);

            if (Object.keys(store.state.recentErrors).length >= 2) return;

            const id = generateId();
            store.state.recentErrors[id] = err;
            window.setTimeout(() => {
                delete store.state.recentErrors[id];
            }, 25000);
        }
    }
});

export const errorHandlerStoreKey: InjectionKey<Store<ErrorHandlerStore>> = Symbol();

export function useErrorHandlerStore(): Store<ErrorHandlerStore> {
    return useStore(errorHandlerStoreKey);
}

export function setUpErrorHandler(app: App<unknown>): void {
    app.use(errorHandlerStore, errorHandlerStoreKey);
    app.config.errorHandler = (err, vm, info) => {
        errorHandlerStore.dispatch('handleError', { err, info });
    };
}
