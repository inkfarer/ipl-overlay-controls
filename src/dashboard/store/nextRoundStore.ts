import { NextRound } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { SetRoundRequest } from 'types/messages/rounds';

const nextRound = nodecg.Replicant<NextRound>('nextRound');

export const nextRoundReps = [ nextRound ];

export interface NextRoundStore {
    nextRound: NextRound;
}

export const nextRoundStore = createStore<NextRoundStore>({
    state: {
        nextRound: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        },
        setShowOnStream(store, newValue: boolean) {
            nextRound.value.showOnStream = newValue;
        }
    },
    actions: {
        beginNextMatch(): void {
            nodecg.sendMessage('beginNextMatch');
        },
        setNextRound(store, {
            teamAId,
            teamBId,
            roundId
        }: { teamAId: string, teamBId: string, roundId: string }): void {
            nodecg.sendMessage('setNextRound', { teamAId, teamBId, roundId } as SetRoundRequest);
        },
    }
});

export const nextRoundStoreKey: InjectionKey<Store<NextRoundStore>> = Symbol();

export function useNextRoundStore(): Store<NextRoundStore> {
    return useStore(nextRoundStoreKey);
}
