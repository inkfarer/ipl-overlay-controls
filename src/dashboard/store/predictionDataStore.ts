import { NodeCGBrowser } from 'nodecg/browser';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { Prediction, PredictionStore } from 'schemas';
import { PredictionStatus } from 'types/enums/predictionStatus';

export const predictionStore = nodecg.Replicant<PredictionStore>('predictionStore');

export const predictionReps = [ predictionStore ];

export interface PredictionDataStore {
    predictionStore: PredictionStore;
}

export const predictionDataStore = createStore<PredictionDataStore>({
    state: {
        predictionStore: null,
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        },
    },
    actions: {
        async lockPrediction(store) {
            return patchPrediction(
                store.state.predictionStore.currentPrediction,
                PredictionStatus.LOCKED,
                [ PredictionStatus.ACTIVE ]);
        },
        async cancelPrediction(store) {
            return patchPrediction(
                store.state.predictionStore.currentPrediction,
                PredictionStatus.CANCELED,
                [ PredictionStatus.ACTIVE, PredictionStatus.LOCKED ]);
        },
        async resolvePrediction(store, { winningOutcomeIndex }: { winningOutcomeIndex: number }) {
            if (winningOutcomeIndex > 1 || winningOutcomeIndex < 0) {
                throw new Error(`Cannot resolve prediction with outcome index ${winningOutcomeIndex}.`);
            } else if (!store.state.predictionStore.currentPrediction?.id) {
                throw new Error('No prediction to resolve.');
            } else if (store.state.predictionStore.currentPrediction.status !== PredictionStatus.LOCKED ) {
                throw new Error('Can only resolve a locked prediction.');
            }

            return nodecg.sendMessage('patchPrediction', {
                id: store.state.predictionStore.currentPrediction.id,
                status: PredictionStatus.RESOLVED,
                winning_outcome_id: store.state.predictionStore.currentPrediction.outcomes[winningOutcomeIndex].id
            });
        },
        async createPrediction(
            store,
            { title, teamAName, teamBName, duration }:
                { title: string, teamAName: string, teamBName: string, duration: number }
        ) {
            if (
                [PredictionStatus.ACTIVE, PredictionStatus.LOCKED]
                    .includes(store.state.predictionStore.currentPrediction?.status as PredictionStatus)
            ) {
                throw new Error('An unresolved prediction already exists.');
            }

            return nodecg.sendMessage('postPrediction', {
                title,
                outcomes: [
                    { title: teamAName },
                    { title: teamBName }
                ],
                prediction_window: duration
            });
        },
        async reconnect() {
            return nodecg.sendMessage('reconnectToRadiaSocket');
        }
    }
});

async function patchPrediction(
    currentPrediction: Prediction,
    status: PredictionStatus,
    allowedForStatuses: PredictionStatus[]
): Promise<void> {
    if (!currentPrediction || !allowedForStatuses.includes(currentPrediction?.status as PredictionStatus)) {
        throw new Error(`Cannot set prediction status at this time. Status must match one of [${allowedForStatuses.join(', ')}]`);
    }

    return nodecg.sendMessage('patchPrediction', {
        id: currentPrediction.id,
        status: status,
    });
}

export const predictionDataStoreKey: InjectionKey<Store<PredictionDataStore>> = Symbol();

export function usePredictionDataStore(): Store<PredictionDataStore> {
    return useStore(predictionDataStoreKey);
}
