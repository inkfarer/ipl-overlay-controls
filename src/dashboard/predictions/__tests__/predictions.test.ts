import Predictions from '../predictions.vue';
import { config, mount } from '@vue/test-utils';
import { createStore } from 'vuex';
import { PredictionDataStore, predictionDataStoreKey } from '../../store/predictionDataStore';
import { PredictionStatus } from 'types/enums/predictionStatus';
import { mockDialog, mockGetDialog, mockSendMessage } from '../../__mocks__/mockNodecg';

describe('Predictions', () => {
    const mockLockPrediction = jest.fn();
    const mockCancelPrediction = jest.fn();

    config.global.stubs = {
        PredictionDataDisplay: true,
        IplDataRow: true,
        IplButton: true,
        IplErrorDisplay: true
    };

    const createPredictionDataStore = () => {
        return createStore<PredictionDataStore>({
            state: {
                predictionStore: {
                    status: {
                        socketOpen: true,
                        predictionsEnabled: true
                    },
                    currentPrediction: {
                        id: 'prediction123',
                        broadcasterId: 'ipl',
                        broadcasterName: 'IPL',
                        broadcasterLogin: 'eye pee el',
                        title: 'Who will win?',
                        outcomes: [
                            {
                                id: 'outcome-1',
                                title: 'First Team',
                                users: 5,
                                pointsUsed: 10000,
                                topPredictors: [],
                                color: 'BLUE'
                            },
                            {
                                id: 'outcome-2',
                                title: 'Second Team',
                                users: 1,
                                pointsUsed: 1,
                                topPredictors: [],
                                color: 'PINK'
                            }
                        ],
                        duration: 60,
                        status: PredictionStatus.ACTIVE,
                        creationTime: '2020'
                    }
                }
            },
            actions: {
                lockPrediction: mockLockPrediction,
                cancelPrediction: mockCancelPrediction,
                reconnect: jest.fn()
            }
        });
    };

    it('shows warning if predictions are not enabled', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.status.predictionsEnabled = false;
        store.state.predictionStore.status.predictionStatusReason = 'Predictions are disabled!';
        store.state.predictionStore.currentPrediction = undefined;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        expect(wrapper.findComponent('[data-test="no-prediction-data-message"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="prediction-data-display"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="prediction-management-space"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="socket-closed-message"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="socket-reconnect-button"]').exists()).toEqual(false);
        const predictionsDisabledMessage = wrapper.findComponent('[data-test="predictions-disabled-message"]');
        expect(predictionsDisabledMessage.isVisible()).toEqual(true);
        expect(predictionsDisabledMessage.text()).toEqual('Predictions are disabled!');
    });

    it('shows message and management space if prediction data is missing', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction = undefined;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        expect(wrapper.findComponent('[data-test="predictions-disabled-message"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="socket-closed-message"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="socket-reconnect-button"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="prediction-data-display"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="no-prediction-data-message"]').isVisible()).toEqual(true);
        expect(wrapper.findComponent('[data-test="prediction-management-space"]').isVisible()).toEqual(true);
    });

    it('shows prediction data if predictions are enabled and data is present', () => {
        const store = createPredictionDataStore();
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        expect(wrapper.findComponent('[data-test="no-prediction-data-message"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="socket-closed-message"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="socket-reconnect-button"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="predictions-disabled-message"]').exists()).toEqual(false);
        expect(wrapper.findComponent('[data-test="prediction-data-display"]').isVisible()).toEqual(true);
        expect(wrapper.findComponent('[data-test="prediction-management-space"]').isVisible()).toEqual(true);
    });

    it('shows message if websocket is closed', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.status.socketOpen = false;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        const message = wrapper.findComponent('[data-test="socket-closed-message"]');
        expect(message.exists()).toEqual(true);
        expect(message.isVisible()).toEqual(true);
        const reconnectButton = wrapper.findComponent('[data-test="socket-reconnect-button"]');
        expect(reconnectButton.exists()).toEqual(true);
        expect(reconnectButton.isVisible()).toEqual(true);
    });

    it('dispatches to store on reconnect', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.status.socketOpen = false;
        jest.spyOn(store, 'dispatch');
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        wrapper.getComponent('[data-test="socket-reconnect-button"]').vm.$emit('click');

        expect(store.dispatch).toHaveBeenCalledWith('reconnect');
    });

    it('shows expected buttons if prediction status is RESOLVED', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.RESOLVED;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        expect(wrapper.get('.prediction-button-container').html()).toMatchSnapshot();
    });

    it('shows expected buttons if prediction status is LOCKED', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        expect(wrapper.get('.prediction-button-container').html()).toMatchSnapshot();
    });

    it('shows expected buttons if prediction status is CANCELED', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.CANCELED;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        expect(wrapper.get('.prediction-button-container').html()).toMatchSnapshot();
    });

    it('shows expected buttons if prediction status is ACTIVE', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.ACTIVE;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        expect(wrapper.get('.prediction-button-container').html()).toMatchSnapshot();
    });

    it('shows dialog when resolving prediction', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        wrapper.getComponent('[data-test="resolve-prediction-button"]').vm.$emit('click');

        expect(mockDialog.open).toHaveBeenCalled();
        expect(mockGetDialog).toHaveBeenCalledWith('resolvePredictionDialog');
    });

    it('shows dialog when creating prediction', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.RESOLVED;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        wrapper.getComponent('[data-test="create-prediction-button"]').vm.$emit('click');

        expect(mockDialog.open).toHaveBeenCalled();
        expect(mockGetDialog).toHaveBeenCalledWith('createPredictionDialog');
    });

    it('dispatches action when locking prediction', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.ACTIVE;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        wrapper.getComponent('[data-test="lock-prediction-button"]').vm.$emit('click');

        expect(mockLockPrediction).toHaveBeenCalled();
    });

    it('dispatches action when cancelling prediction', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        wrapper.getComponent('[data-test="cancel-prediction-button"]').vm.$emit('click');

        expect(mockCancelPrediction).toHaveBeenCalled();
    });

    it('sends message when showing prediction data', () => {
        const store = createPredictionDataStore();
        store.state.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        const wrapper = mount(Predictions, {
            global: {
                plugins: [[store, predictionDataStoreKey]]
            }
        });

        wrapper.getComponent('[data-test="show-prediction-button"]').vm.$emit('click');

        expect(mockSendMessage).toHaveBeenCalledWith('showPredictionData');
    });
});
