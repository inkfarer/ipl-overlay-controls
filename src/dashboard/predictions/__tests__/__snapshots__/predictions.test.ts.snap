// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Predictions shows expected buttons if prediction status is ACTIVE 1`] = `"<div class=\\"m-t-6 prediction-button-container\\"><!--v-if--><!--v-if--><ipl-button-stub label=\\"Lock\\" color=\\"red\\" disabled=\\"false\\" async=\\"true\\" progressmessage=\\"Locking...\\" successmessage=\\"Done!\\" requiresconfirmation=\\"true\\" shortconfirmationmessage=\\"true\\" small=\\"false\\" disableonsuccess=\\"false\\" data-test=\\"lock-prediction-button\\"></ipl-button-stub><ipl-button-stub label=\\"Cancel\\" color=\\"red\\" disabled=\\"false\\" async=\\"true\\" progressmessage=\\"Cancelling...\\" successmessage=\\"Done!\\" requiresconfirmation=\\"true\\" shortconfirmationmessage=\\"true\\" small=\\"false\\" disableonsuccess=\\"false\\" data-test=\\"cancel-prediction-button\\"></ipl-button-stub><ipl-button-stub label=\\"Show\\" color=\\"blue\\" disabled=\\"false\\" async=\\"false\\" progressmessage=\\"Working...\\" successmessage=\\"Sent!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"true\\" data-test=\\"show-prediction-button\\"></ipl-button-stub></div>"`;

exports[`Predictions shows expected buttons if prediction status is CANCELED 1`] = `"<div class=\\"m-t-6 prediction-button-container\\"><!--v-if--><ipl-button-stub label=\\"New\\" color=\\"green\\" disabled=\\"false\\" async=\\"false\\" progressmessage=\\"Working...\\" successmessage=\\"Done!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"false\\" data-test=\\"create-prediction-button\\"></ipl-button-stub><!--v-if--><!--v-if--><ipl-button-stub label=\\"Show\\" color=\\"blue\\" disabled=\\"false\\" async=\\"false\\" progressmessage=\\"Working...\\" successmessage=\\"Sent!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"true\\" data-test=\\"show-prediction-button\\"></ipl-button-stub></div>"`;

exports[`Predictions shows expected buttons if prediction status is LOCKED 1`] = `"<div class=\\"m-t-6 prediction-button-container\\"><ipl-button-stub label=\\"Resolve\\" color=\\"green\\" disabled=\\"false\\" async=\\"false\\" progressmessage=\\"Working...\\" successmessage=\\"Done!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"false\\" data-test=\\"resolve-prediction-button\\"></ipl-button-stub><!--v-if--><!--v-if--><ipl-button-stub label=\\"Cancel\\" color=\\"red\\" disabled=\\"false\\" async=\\"true\\" progressmessage=\\"Cancelling...\\" successmessage=\\"Done!\\" requiresconfirmation=\\"true\\" shortconfirmationmessage=\\"true\\" small=\\"false\\" disableonsuccess=\\"false\\" data-test=\\"cancel-prediction-button\\"></ipl-button-stub><ipl-button-stub label=\\"Show\\" color=\\"blue\\" disabled=\\"false\\" async=\\"false\\" progressmessage=\\"Working...\\" successmessage=\\"Sent!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"true\\" data-test=\\"show-prediction-button\\"></ipl-button-stub></div>"`;

exports[`Predictions shows expected buttons if prediction status is RESOLVED 1`] = `"<div class=\\"m-t-6 prediction-button-container\\"><!--v-if--><ipl-button-stub label=\\"New\\" color=\\"green\\" disabled=\\"false\\" async=\\"false\\" progressmessage=\\"Working...\\" successmessage=\\"Done!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"false\\" data-test=\\"create-prediction-button\\"></ipl-button-stub><!--v-if--><!--v-if--><ipl-button-stub label=\\"Show\\" color=\\"blue\\" disabled=\\"false\\" async=\\"false\\" progressmessage=\\"Working...\\" successmessage=\\"Sent!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"true\\" data-test=\\"show-prediction-button\\"></ipl-button-stub></div>"`;
